feat: add initial eventhubs test components (#265)
arcus-azure/arcus.testing: docs/preview/03-Features/04-Azure/05-Messaging/02-eventhubs.md

# Event Hubs

> The `Arcus.Testing.Messaging.EventHubs` package provides test fixtures related to Azure Event Hubs. By using the common testing practice 'clean environment', it provides a temporary hub.

Reviewer: ```suggestion
The `Arcus.Testing.Messaging.EventHubs` package provides test fixtures related to Azure Event Hubs.  It adheres to the 'clean environment' testing practice by providing a temporary Event Hub.
```

## Installation
The following functionality is available when installing this package:

```powershell
PM> Install-Package -Name Arcus.Testing.Messaging.EventHubs
```

## Temporary hub
The `TemporaryEventHub` provides a solution when the integration test requires an Azure Event Hub during the test run. The test fixture creates a hub upon the setup of the fixture and deletes it again when the fixture disposes.

> > ✨ Only when the test fixture was responsible for creating the hub, will the fixture delete the hub upon disposal. This follows the 'clean environment' testing principle that describes that after the test run, the test should revert the state back to its original state.

Reviewer: ```suggestion
> ✨ When the test fixture was responsible for creating the hub, the fixture will delete the hub when the test disposes the fixture. This follows the 'clean environment' testing principle that describes that a test should revert the environment to its original state after the test has run.
```

Author: 'when the test has finished' might not always be when the 'test fixture disposes'. In most cases: yes, but that is more related to how the test was implemented. This remark only talks about the lifetime of the test fixture, independent on how it is used.
Plus, without mentioning 'dispose', I would imagine people thinking that it is some kind of 'aspect-oriented' stuff that gets hooked in.

Reviewer: Ok, then change 'when the test has finished' to 'when the test has disposed' :)

Author: 'Test disposes the fixture'? Added this suggestion.
